### ISSUE TITLE
Black screen after login. `gnome-shell-extension-cosmic-dock` uninstallable after upgrade.

### ISSUE THREAD

**Issue by leorochael** [pop-os/pop#2635]
**Distribution (run `cat /etc/os-release`):**
 
This is a machine that was recently upgraded from 20.04 to 22.04.

 ```
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

```
# apt policy pop-session 
pop-session:
  Installed: 4.0.6~1612395801~20.04~5367ce8
  Candidate: 5.0.0~1662569841~22.04~1c78668
  Version table:
     5.0.0~1662569841~22.04~1c78668 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
       1001 http://apt.pop-os.org/release jammy/main i386 Packages
 *** 4.0.6~1612395801~20.04~5367ce8 100
        100 /var/lib/dpkg/status
```

Also of interest:

```
root@pop-os:~# apt full-upgrade 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following packages have been kept back:
  pop-session
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.


root@pop-os:~# apt install pop-session
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 pop-session : Depends: gnome-shell-extension-cosmic-dock but it is not installable
E: Unable to correct problems, you have held broken packages.


root@pop-os:~# apt install pop-session gnome-shell-extension-cosmic-dock
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package gnome-shell-extension-cosmic-dock is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'gnome-shell-extension-cosmic-dock' has no installation candidate
root@pop-os:~# 

root@pop-os:~# for i in /etc/apt/sources.list /etc/apt/sources.list.d/* ; do [[ $i == *.save ]] && continue ; echo -e '\n*** FILENAME:' $i && cat $i ; done

*** FILENAME: /etc/apt/sources.list
## This file is deprecated in Pop!_OS.
## See `man deb822` and /etc/apt/sources.list.d/system.sources.

*** FILENAME: /etc/apt/sources.list.d/google-chrome.list
### THIS FILE IS AUTOMATICALLY CONFIGURED ###
# You may comment out this entry, but any other modifications may be lost.
# deb [arch=amd64] https://dl.google.com/linux/chrome/deb/ stable main

*** FILENAME: /etc/apt/sources.list.d/pop-os-apps.sources
X-Repolib-Name: Pop_OS Apps
Enabled: yes
Types: deb
URIs: http://apt.pop-os.org/proprietary
Suites: jammy
Components: main

*** FILENAME: /etc/apt/sources.list.d/pop-os-release.sources
X-Repolib-Name: Pop_OS Release Sources
Enabled: yes
Types: deb deb-src
URIs: http://apt.pop-os.org/release
Suites: jammy
Components: main

*** FILENAME: /etc/apt/sources.list.d/system.sources
X-Repolib-Name: Pop_OS System Sources
Enabled: yes
Types: deb deb-src
URIs: http://apt.pop-os.org/ubuntu
Suites: jammy jammy-security jammy-updates jammy-backports
Components: main restricted universe multiverse
X-Repolib-Default-Mirror: http://apt.pop-os.org/ubuntu
X-Repolib-ID: system

```

**Issue/Bug Description:**

After upgrading from pop.os 22.04 and logging in, I had no mouse pointer and had to navigate the first-run wizard by keyboard. And after that wizard closed, I had a black screen with no interactivity whatsoever. Rebooting from a virtual console and logging in again went straight to the black screen with no mouse pointer.

Following [these instructions](https://support.system76.com/articles/login-loop-pop/) I created a new user but that only repeated the same situation: black screen with no mouse pointer after login (except for the first run wizard on first login).

I've managed to regain graphical control of my machine by installing `gnome-session` with `apt` and selecting the `GNOME` session on the login screen.

**Steps to reproduce (if you know):**

Upgrade from Pop!OS 22.04

**Expected behavior:**

The pop desktop session should work normally.

**Other Notes:**

I cannot prove, but highly suspect, that the lack of `gnome-shell-extension-cosmic-dock` is what is causing the pop-os session to misbehave. Which repository is supposed to carry this package?

I suggest adding the tip about installing the `gnome-session` package to the [Black Screen or Login Issues](https://support.system76.com/articles/login-loop-pop/) guide.

**Comment by Adam-Kadmon** [comment]
I know this kind of "me too" comment isn't usually helpful, but I ran into the same problem, also when upgrading from 20.04 to 22.04 and also a couple of days ago.

I see that `pop-session` is held back (to the 20.04 version) on my machine. I get the same dependency problems with gnome-`shell-extension-cosmic-dock`, which is also uninstallable on my system.

I tried everything in the Black Screen or Login Issues guide, but to no avail.

Thank you for the `gnome-session` workaround - it's a good backup plan in case I end up needing a GUI. I'll try unholding `pop-session` tonight to see if that helps. I also haven't tried the manual repair instructions (`pop-upgrade release repair`, `sudo apt-get install -f`, `sudo apt-get full-upgrade --allow-downgrades`) yet.

I'll report back tonight.
